pkg/reconciler/awssnssource/adapter.go

```diff
@@ -34,27 +34,49 @@ import (
 
 const adapterName = "awssnssource"
 
+const (
+	envSubscriptionAttrs = "SUBSCRIPTION_ATTRIBUTES"
+	envPublicURL         = "PUBLIC_URL"
+)
+
 // adapterConfig contains properties used to configure the source's adapter.
 // These are automatically populated by envconfig.
 type adapterConfig struct {
 	// Container image
 	Image string `default:"gcr.io/triggermesh/awssnssource"`
+
+	// Public domain used to expose Knative Services
+	Domain string `envconfig:"KNATIVE_DOMAIN" default:"example.com"`
+	// URL scheme used to expose public Knative Services
+	Scheme string `envconfig:"KNATIVE_URL_SCHEME" default:"http"`
+
 	// Configuration accessor for logging/metrics/tracing
 	configs source.ConfigAccessor
 }
 
-// adapterDeploymentBuilder returns an AdapterDeploymentBuilderFunc for the
+// adapterServiceBuilder returns an AdapterServiceBuilderFunc for the
 // given source object and adapter config.
-func adapterDeploymentBuilder(src *v1alpha1.AWSSNSSource, cfg *adapterConfig) common.AdapterDeploymentBuilderFunc {
-	return func(arn arn.ARN, sinkURI *apis.URL) *appsv1.Deployment {
+func adapterServiceBuilder(src *v1alpha1.AWSSNSSource, cfg *adapterConfig) common.AdapterServiceBuilderFunc {
+	return func(arn arn.ARN, sinkURI *apis.URL) *servingv1.Service {
 		name := kmeta.ChildName(fmt.Sprintf("%s-", adapterName), src.Name)
 
 		var sinkURIStr string
 		if sinkURI != nil {
 			sinkURIStr = sinkURI.String()
 		}
 
-		return resource.NewDeployment(src.Namespace, name,
+		var subsAttrsStr string
+		if subsAttrsJSON, err := json.Marshal(src.Spec.SubscriptionAttributes); err == nil {
+			subsAttrsStr = string(subsAttrsJSON)
+		}
+
+		/* FIXME(antoineco): we wouldn't need to know the service URL in
+		   advance if we could reconcile SNS subscriptions from the controller.
+		   Ref. https://github.com/triggermesh/aws-event-sources/issues/157
+		*/
+		adapterURL := fmt.Sprintf("%s://%s.%s.%s", cfg.Scheme, name, src.Namespace, cfg.Domain)
```

Comment:
This is a terrible thing to do. Unfortunately Pods can not introspect their own public URL. We wouldn't have to resort to such hack if we had a solution for #157.